crates/superposition_types/src/config.rs

```diff
@@ -0,0 +1,259 @@
+    }
+}
+
+impl IntoIterator for Overrides {
+    type Item = (String, Value);
+    type IntoIter = <Map<String, Value> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl_try_from_map!(Cac, Overrides, Overrides::validate_data);
+impl_try_from_map!(Exp, Overrides, Overrides::validate_data);
+
+#[cfg(feature = "diesel_derives")]
+#[derive(
+    Deserialize,
+    Serialize,
+    Clone,
+    AsRef,
+    Deref,
+    Debug,
+    Eq,
+    PartialEq,
+    Into,
+    AsExpression,
+    FromSqlRow,
+    JsonFromSql,
+    JsonToSql,
+)]
+#[diesel(sql_type = Json)]
+pub struct Condition(Map<String, Value>);
+
+#[cfg(not(feature = "diesel_derives"))]
+#[derive(Deserialize, Serialize, Clone, AsRef, Deref, Debug, Eq, PartialEq, Into)]
+pub struct Condition(Map<String, Value>);
+
+impl Condition {
+    fn validate_data_for_cac(condition_map: Map<String, Value>) -> Result<Self, String> {
+        if condition_map.is_empty() {
+            log::error!("Condition validation error: Context is empty");
+            return Err("Context should not be empty".to_owned());
+        }
+        jsonlogic::expression::Expression::from_json(&json!(condition_map)).map_err(
+            |msg| {
+                log::error!("Condition validation error: {}", msg);
+                msg
+            },
+        )?;
+        Ok(Self(condition_map))
+    }
+
+    fn validate_data_for_exp(condition_map: Map<String, Value>) -> Result<Self, String> {
+        let condition_val = json!(condition_map);
+        let ast = jsonlogic::expression::Expression::from_json(&condition_val).map_err(
+            |msg| {
+                log::error!("Condition validation error: {}", msg);
+                msg
+            },
+        )?;
+        let dimensions = ast.get_variable_names().map_err(|msg| {
+            log::error!("Error while parsing variable names : {}", msg);
+            msg
+        })?;
+        if dimensions.contains("variantIds") {
+            log::error!("experiment's context should not contain variantIds dimension");
+            return Err(
+                "experiment's context should not contain variantIds dimension"
+                    .to_string(),
+            );
+        }
+        Ok(Self(condition_map))
+    }
+}
+
+impl_try_from_map!(Cac, Condition, Condition::validate_data_for_cac);
+impl_try_from_map!(Exp, Condition, Condition::validate_data_for_exp);
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Context {
+    pub id: String,
+    pub condition: Condition,
+    pub priority: i32,
+    pub override_with_keys: [String; 1],
+}
+
+impl Contextual for Context {
+    fn get_condition(&self) -> Condition {
+        self.condition.clone()
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+pub struct Config {
+    pub contexts: Vec<Context>,
+    pub overrides: HashMap<String, Overrides>,
+    pub default_configs: Map<String, Value>,
+}
```

Comment:
INFO: Common `Config` and `Context` types which are now used across all crates which require these types